tests/unit/multiclass/tree/RandomForest_unittest.cc

```diff
@@ -279,3 +282,126 @@ TEST(RandomForest,classify_non_nominal_test)
 	SG_UNREF(c);
 	SG_UNREF(eval);
 }
+
+TEST(RandomForest, test_probabilities)
+{
+	sg_rand->set_seed(1);
+
+	SGMatrix<float64_t> data(4, 14);
+	SGVector<float64_t> lab(14);
+
+	generate_nm_data(data, lab);
+
+	CDenseFeatures<float64_t>* feats = new CDenseFeatures<float64_t>(data);
+
+	SGVector<bool> ft = SGVector<bool>(4);
+	ft[0] = true;
+	ft[1] = true;
+	ft[2] = true;
+	ft[3] = true;
+
+	CMulticlassLabels* labels = new CMulticlassLabels(lab);
+
+	CRandomForest* c = new CRandomForest(feats, labels, 100, 2);
+	c->set_feature_types(ft);
+	CMajorityVote* mv = new CMajorityVote();
+	c->set_combination_rule(mv);
+	c->parallel->set_num_threads(1);
+	c->train(feats);
+
+	SGMatrix<float64_t> test(4, 5);
+	test(0, 0) = overcast;
+	test(0, 1) = rain;
+	test(0, 2) = sunny;
+	test(0, 3) = rain;
+	test(0, 4) = sunny;
+
+	test(1, 0) = hot;
+	test(1, 1) = cool;
+	test(1, 2) = mild;
+	test(1, 3) = mild;
+	test(1, 4) = hot;
+
+	test(2, 0) = normal;
+	test(2, 1) = high;
+	test(2, 2) = high;
+	test(2, 3) = normal;
+	test(2, 4) = normal;
+
+	test(3, 0) = strong;
+	test(3, 1) = strong;
+	test(3, 2) = weak;
+	test(3, 3) = weak;
+	test(3, 4) = strong;
+
+	CDenseFeatures<float64_t>* test_feats = new CDenseFeatures<float64_t>(test);
+
+	CBinaryLabels* result = c->apply_binary(test_feats);
+	SGVector<float64_t> res_vector = result->get_labels();
+	SGVector<float64_t> values_vector = result->get_values();
+
+	EXPECT_FLOAT_EQ(0.83, values_vector[0]);
+	EXPECT_FLOAT_EQ(0.359999999999, values_vector[1]);
+	EXPECT_FLOAT_EQ(0.27, values_vector[2]);
+	EXPECT_FLOAT_EQ(0.91, values_vector[3]);
+	EXPECT_FLOAT_EQ(0.53, values_vector[4]);
+
+	CMulticlassAccuracy* eval = new CMulticlassAccuracy();
+	EXPECT_NEAR(0.642857, c->get_oob_error(eval), 1e-6);
+
+	SG_UNREF(test_feats);
+	SG_UNREF(result);
+	SG_UNREF(c);
+	SG_UNREF(eval);
+}
+
+TEST(RandomForest, test_output)
+{
+	sg_rand->set_seed(1);
+	float64_t data_A[] = {-2.0, -1.0, -1.0, 1.0, 1.0, 2.0,
+	                      -1.0, -1.0, -2.0, 1.0, 2.0, 1.0};
+
+	SGMatrix<float64_t> data(data_A, 2, 6, false);
+
+	CDenseFeatures<float64_t>* features_train =
+	    new CDenseFeatures<float64_t>(data);
+
+	float64_t labels[] = {0.0, 0.0, 0.0, 1.0, 1.0, 1.0};
+	SGVector<float64_t> lab(labels, 6);
+	CMulticlassLabels* labels_train = new CMulticlassLabels(lab);
+
+	CRandomForest* c = new CRandomForest(features_train, labels_train, 10, 2);
+	SGVector<bool> ft = SGVector<bool>(2);
+	ft[0] = true;
+	ft[1] = true;
+	c->set_feature_types(ft);
+
+	CMeanRule* mr = new CMeanRule();
+	c->set_combination_rule(mr);
+	c->parallel->set_num_threads(1);
```

Comment:
I didn't write these tests, but by changing the number of threads set to 1, the assertion for the oob error fails (exceed 1e-6 tolerance).